### ISSUE TITLE
Hotkey release is not always handled.

### ISSUE THREAD

**Issue by Xylot** [geheur/More-menu-entry-swaps#8]
I encountered this while using Runelite's quick world hopper (CTRL-Shift-(L/R) Arrow). The hotkey action stays in effect without the key pressed after quick hopping if one of the keys bound is CTRL/Shift/Arrow.

**Example:** Withdraw-1 is set to CTRL. After hopping to the next world (using CTRL-Shift-Right Arrow), the withdraw option is still on withdraw-1 without CTRL pressed. Pressing CTRL again releases/resets it.

**Solution:** Probably just clearing any hotkey press on-login.

**Comment by geheur** [comment]
This doesn't seem to have anything to do with hopping world but with pressing the following key sequence:
ctrl down
shift down
ctrl up
shift up

This is probably something to do with how modifier keys are handled.

**Comment by Xylot** [comment]
Thank you for the title change, it makes much more sense now. What layer do you believe this issue is sourced from? OS, Runelite, or this project?

**Comment by geheur** [comment]
I don't need to press the arrow keys at all to make this bug happen, just what I wrote. If I world hop with [ctrl down, shift down, rightarrow down, rightarrow up, shift up, ctrl up], I don't get this bug so I don't think it has anything to do with arrow keys.

I think the issue is my plugin doesn't recognize ctrl up with shift held down as ctrl up.

**Comment by m00fin** [comment]
Just wanted to add that this issue occurs in a number of other situations in my experience, as well as in other plugins that accept keybind assignments. I haven't really been taking note of exactly when so unfortunately cannot provide exact details.

Also:

- I've noticed that this issue occurs on OSX with the Mac specific modifiers (cmd/option) when used in some ctrl/shift combinations as well.
- I directly copied my settings file from a device running Linux to a device running OSX and any bindings that used ctrl + <key> did not work. I was able to assign ctrl + <key>, however none would work and I had to change the bindings to use a modifier that was not ctrl.

I will begin taking better notice of these issues and begin giving more detailed reports and dig through the client source if I have the time as it may be more appropriate to report them there as this seems to be an issue in other plugins as well.
